pageserver/src/metrics.rs

```diff
@@ -86,11 +86,20 @@ pub(crate) static STORAGE_TIME_GLOBAL: Lazy<HistogramVec> = Lazy::new(|| {
     .expect("failed to define a metric")
 });
 
-pub(crate) static READ_NUM_FS_LAYERS: Lazy<Histogram> = Lazy::new(|| {
+pub(crate) static READ_NUM_LAYERS_VISITED: Lazy<Histogram> = Lazy::new(|| {
     register_histogram!(
-        "pageserver_read_num_fs_layers",
-        "Number of persistent layers accessed for processing a read request, including those in the cache",
-        vec![1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 10.0, 20.0, 50.0, 100.0],
+        "pageserver_layers_visited_per_read_global",
+        "Number of layers visited to reconstruct one key",
+        vec![1.0, 4.0, 8.0, 16.0, 32.0, 64.0, 128.0, 256.0, 512.0, 1024.0],
```

Comment:
I would prefer to have the default image_layer_creation_threshold or whatever it was called here, as we now the old 10 would be split between 8 and 16, which feels a bit much considering the normal would be maybe we can only run compaction at 15 L0s..

Is there a reason why the default threshold value shouldn't be included here, perhaps my understanding of the code is wrong?

Reply:
I didn't include it because under load it seems like we shoot above it by quite a bit. The average in the test with very aggresive compaction was still around 7. I can change the `4.0` to a `3.0` if you prefer.

> which feels a bit much

Not sure what you mean. Is it too granular or not granular enough in your view?

Reply:
> I am just wondering if we are able to see the difference between "this tenant often gets late compaction" and "this other tenant seems to be working perfectly".

I think we should define a separate metric for this (this one depends on reads actually happening). We could define compaction lag as `last_record_lsn - last_compacted_lsn`. I think I mentioned this to Christian at some point.

Reply:
Let's merge and see how it goes. I'll be watching this to figure out a good value for the logging threshold.